src/main/java/games/strategy/triplea/oddsCalculator/ta/OddsCalculator.java

```diff
@@ -178,25 +167,34 @@ private AggregateResults calculate(final int count) {
         OddsCalculator.getUnitListByOrderOfLoss(this.attackerOrderOfLosses, attackingUnits, gameData);
     final List<Unit> defenderOrderOfLosses =
         OddsCalculator.getUnitListByOrderOfLoss(this.defenderOrderOfLosses, defendingUnits, gameData);
-    for (int i = 0; i < count && !cancelled; i++) {
-      final CompositeChange allChanges = new CompositeChange();
-      final DummyDelegateBridge bridge1 =
-          new DummyDelegateBridge(attacker, gameData, allChanges, attackerOrderOfLosses, defenderOrderOfLosses,
-              keepOneAttackingLandUnit, retreatAfterRound, retreatAfterXUnitsLeft, retreatWhenOnlyAirLeft);
-      final GameDelegateBridge bridge = new GameDelegateBridge(bridge1);
-      final MustFightBattle battle = new MustFightBattle(location, attacker, gameData, battleTracker);
-      battle.setHeadless(true);
-      battle.isAmphibious();
-      battle.setUnits(defendingUnits, attackingUnits, bombardingUnits,
-          (amphibious ? attackingUnits : new ArrayList<>()), defender, territoryEffects);
-      bridge1.setBattle(battle);
-      battle.fight(bridge);
-      aggregateResults.addResult(new BattleResults(battle, gameData));
-      // restore the game to its original state
-      gameData.performChange(allChanges.invert());
-      battleTracker.clear();
-      battleTracker.clearBattleRecords();
-    }
+    final AggregateResults aggregateResults = IntStream.range(0, count).parallel()
+        .filter(i -> !cancelled)
+        .mapToObj(i -> {
+          final CompositeChange allChanges = new CompositeChange();
+          final BattleTracker battleTracker = new BattleTracker();
+          try {
+            final DummyDelegateBridge bridge1 =
+                new DummyDelegateBridge(attacker, gameData, allChanges, attackerOrderOfLosses, defenderOrderOfLosses,
+                    keepOneAttackingLandUnit, retreatAfterRound, retreatAfterXUnitsLeft, retreatWhenOnlyAirLeft);
+            final GameDelegateBridge bridge = new GameDelegateBridge(bridge1);
+            final MustFightBattle battle = new MustFightBattle(location, attacker, gameData, battleTracker);
+            battle.setHeadless(true);
+            battle.isAmphibious();
+            battle.setUnits(defendingUnits, attackingUnits, bombardingUnits,
+                (amphibious ? attackingUnits : new ArrayList<>()), defender, territoryEffects);
+            bridge1.setBattle(battle);
+            battle.fight(bridge);
+            return new BattleResults(battle, gameData);
+          } finally {
+            // restore the game to its original state
+            gameData.performChange(allChanges.invert());
+            battleTracker.clear();
+            battleTracker.clearBattleRecords();
+          }
+        }).collect(Collector.of(() -> new AggregateResults(count), (a, b) -> a.addResult(b), (a1, a2) -> {
```

Comment:
This line is concise, but a bit hard to read.

Formatting to a drop-case style I think helps, makes it a bit easier to parse out the params:
```
        }).collect(Collector.of(
            () -> new AggregateResults(count),
            AggregateResults::addResult,
            (a1, a2) -> {
              a1.addResults(a2.getResults());
              return a1;
            }));
```

I think more ideally the collector would be extracted out to AggregateResults, so we'd have something like:
```
 }).collect(AggregateResults.unionCollector())
```